### PR TITLE
Core component: email

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

Developers can easily integrate their own email providers instead of using Kirby's default PHPMailer library.

### Features
- Core component: email

````php
class CustomEmailProvider extends Kirby\Email\Email
{
    public function __construct(array $props = [], bool $debug = false)
    {
        parent::__construct($props, $debug);
    }

    public function send(bool $debug = false): bool
    {
        // sending stuff
                
        return true;
    }
}

Kirby::plugin('my/email', [
    'components' => [
        'email' => function ($kirby, $props, $debug) {
            return new CustomEmailProvider($props, $debug);
        }
    ]
]);
````

## Breaking changes

None

## Related issues/ideas

- Implements https://kirby.nolt.io/326

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

**Comment by afbora** [comment]
### SendGrid Example

````php
class SendGridProvider extends Kirby\Email\Email
{
    public function __construct(array $props = [], bool $debug = false)
    {
        parent::__construct($props, $debug);
    }

    public function send(): bool
    {
        $sendgrid = new SendGrid("SENDGRID_APIKEY");
        $email    = new SendGrid\Email();

        $email->addTo(current(array_keys($this->to())))
              ->setFrom($this->from())
              ->setSubject($this->subject())
              ->setHtml($this->body()->html());

        return $sendgrid->send($email);
    }
}

Kirby::plugin('my/sendgrid', [
    'components' => [
        'email' => function ($kirby, $props, $debug) {
            return new SendGridProvider($props, $debug);
        }
    ]
]);

**Comment by afbora** [comment]
### Mailgun Example

````php
class MailgunProvider extends Kirby\Email\Email
{
    public function __construct(array $props = [], bool $debug = false)
    {
        parent::__construct($props, $debug);
    }

    public function send(): bool
    {
        $mg = Mailgun\Mailgun::create('key-example'); // For US servers
        $mg = Mailgun\Mailgun::create('key-example', 'https://api.eu.mailgun.net'); // For EU servers

        return $mg->messages()->send('example.com', [
          'from'    => $this->from(),
          'to'      => current(array_keys($this->to())),
          'subject' => $this->subject(),
          'text'    => $this->body()->text()
        ]);
    }
}

Kirby::plugin('my/mailgun', [
    'components' => [
        'email' => function ($kirby, $props, $debug) {
            return new MailgunProvider($props, $debug);
        }
    ]
]);
````

File: config/components.php [hunk 1]
```diff
@@ -55,6 +56,16 @@
         return $output;
     },
 
+    /**
+     * Add your own email provider
+     *
+     * @param array $props
+     * @param bool $debug
+     */
+    'email' => function (array $props = [], bool $debug = false) {
```
Comment (bastianallgeier) [inline config/components.php]: Very cool! I would just pass the Kirby instance as first argument to keep it consistent with every other component.

Comment (afbora) [reply config/components.php]: Nice catch! Fixed 👍

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**